client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -133,9 +134,21 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
 
         final Object bodyContentObject = methodParser.body(args);
         if (bodyContentObject != null) {
-            final String mimeType = "application/json";
-            final String bodyContentString = serializer.serialize(bodyContentObject);
-            request.withBody(bodyContentString, mimeType);
+            final String contentType = methodParser.bodyContentType();
+            if (ContentType.APPLICATION_JSON.equalsIgnoreCase(contentType)) {
+                final String bodyContentString = serializer.serialize(bodyContentObject);
+                request.withBody(bodyContentString, contentType);
+            }
+            else if (bodyContentObject instanceof byte[]) {
+                request.withBody((byte[]) bodyContentObject, contentType == null ? ContentType.APPLICATION_OCTET_STREAM : contentType);
+            }
+            else if (bodyContentObject instanceof String) {
+                request.withBody((String) bodyContentObject, contentType == null ? ContentType.APPLICATION_OCTET_STREAM : contentType);
```

Comment:
Is this the right content type for a string body?

Reply:
I originally had the Content-Type set to "application/json", but that always surrounded the body with double quotes (hello world -> "hello world"). In JSON, that is the correct thing to do, but if we're just sending a String body, then I don't think we'd need to quote it. The real test is if we have any service interfaces that use String bodies. I'm guessing most of them use provider-specific objects, so we may end up removing this branch of the if-statement.